### DISCUSSION TITLE
May I ask how to set the timeout time for raster tile loading？

### DISCUSSION THREAD

**Discussion by fyuanz** [maplibre/maplibre-gl-js#3534] [Q&A] [answered]
Unable to enter the "idle" listener due to prolonged loading ，can I set a short time to ignore tiles that are not loaded after this time？

**Comment by HarelM** [comment, marked as answer]
Hmm... I can't find an easy way to do it.
You can use `addProtocol` to override the fetch behavior and use abort controller to abort after a timeout, but it's very complicated.
Also all the logic around which tile to fetch, cache etc is internal, so I don't think you'll have an easy way to override it.
The following is the relevant method:
https://github.com/maplibre/maplibre-gl-js/blob/39003c93fcef8b6cf06520e0de391042ca1afbb1/src/util/ajax.ts#L141
Not sure I can't think of a good way to pass this information...
Feel free to open a PR if this feature is needed for your case.

**Reply by fyuanz** [reply]
Thank you, it looks so complicated
